### PR TITLE
Update e2e test config for 1.14 release

### PR DESCRIPTION
**Pull request by dtzar**
-->
```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
@dtzar there is one reference to `v1.13.0` that didn't get updated to `v1.14.0` in azure-dev.yaml, line 103. After I changed that, e2e tests run (at least locally).

**Comment by dtzar** [comment]
@mboersma - looks like that didn't fix the e2e tests

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -83,29 +83,29 @@ providers:
...
-    - name: v1.13.0 # latest patch of latest minor in supported v1beta1 releases; this is used for v1beta1 latest --> v1beta1 current clusterctl upgrades test only.
-      value: https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/download/v1.13.0/infrastructure-components.yaml
+    - name: v1.14.0 # latest patch of latest minor in supported v1beta1 releases; this is used for v1beta1 latest --> v1beta1 current clusterctl upgrades test only.
+      value: https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/download/v1.14.0/infrastructure-components.yaml
       type: url
       contract: v1beta1
       files:
       - sourcePath: "../data/shared/v1beta1_provider/metadata.yaml"
-      - sourcePath: "../data/infrastructure-azure/v1.13.0/cluster-template-prow.yaml"
+      - sourcePath: "../data/infrastructure-azure/v1.13.2/cluster-template-prow.yaml"
```
Comment (jackfrancis) [inline test/e2e/config/azure-dev.yaml:103]: Should this be `"../data/infrastructure-azure/v1.14.0/cluster-template-prow.yaml"` ?

**Comment by dtzar** [comment]
Backing out of this PR in favor of https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4641/files

**Pull request closed without merging**